pkg/proxy/handlers/spacelister_get.go

```diff
@@ -77,9 +79,16 @@ func GetUserWorkspace(ctx echo.Context, spaceLister *SpaceLister, workspaceName
 		ctx.Logger().Error(fmt.Sprintf("unauthorized access - there is no SpaceBinding present for the user %s and the workspace %s", userSignup.CompliantUsername, workspaceName))
 		return nil, nil
 	}
+
+	// list all SpaceBindingRequests , just in case there might be some failing to create a SpaceBinding resource.
+	allSpaceBindingRequests, err := listSpaceBindingRequestsForSpace(ctx, GetMembersFunc, space)
```

Comment:
I'm a bit concerned about adding even more stuff to the `GetUserWorkspace()` function. This function is currently used not only in our WS lister API but it's called for every request to the proxy in case there is a workspace context in the request. And we don't care about failed SBRs in that case (or even about the available user roles either btw).
Maybe it's time to split this function to two? One to remain a lightweight version for the general proxy requests and and with all additional information for the WS lister. Or make this function flexible/configurable so it can be used everywhere.

Reply:
Makes sense. Let me take a look an split it so that proxy request do not have to populate the bindings list. Thanks for the suggestion.

Reply:
@alexeykazakov @MatousJobanek  I've created 2 functions  `GetUserWorkspace` used by proxy only and `GetUserWorkspaceWithBindings` used by the spacelister.

Please take a look when you have some time. 

PS: there is one new error condition which is not tested https://github.com/codeready-toolchain/registration-service/pull/387/commits/574d4ebfe6d4c145286ddd6cdf7110d038e128d3#diff-23bf3528567b934bf6ab8d0d690dc61a0ac564ba1b43b89cb5ddde609bb150b1R75-R79 , I can add a specific test in the `proxy_test.go` as I'm not sure where that should go in the current `TestProxy` since it's more a combination of different headers and requests.

Reply:
It's not that important where you add this test. Add it to whatever place you think makes sense. But let's add a test for it if it's not too tricky.

Reply:
I've added a specific test for this error condition in https://github.com/codeready-toolchain/registration-service/pull/387/commits/98d2640e31aeb72c2b54076d7f7fcb6b9bd10c99